js/ui/scheduler/workspaces/view_data_provider.js

```diff
@@ -406,4 +406,53 @@ export default class ViewDataProvider {
         const { groupedData } = this.viewData;
         return groupedData.filter(item => item.groupIndex === groupIndex)[0];
     }
+
+    findGlobalCellPosition(date, groupIndex = 0, allDay = false) {
+        const { completeViewDataMap, _workspace: workspace } = this;
+
+        const showAllDayPanel = workspace._isShowAllDayPanel();
+        const isVerticalGroupOrientation = workspace._isVerticalGroupedWorkSpace();
+
+        for(let rowIndex = 0; rowIndex < completeViewDataMap.length; rowIndex += 1) {
+            const currentRow = completeViewDataMap[rowIndex];
+
+            for(let columnIndex = 0; columnIndex < currentRow.length; columnIndex += 1) {
+                const cellData = currentRow[columnIndex];
+                const {
+                    startDate: currentStartDate,
+                    endDate: currentEndDate,
+                    groupIndex: currentGroupIndex,
+                    allDay: currentAllDay,
+                } = cellData;
+
+                if(groupIndex === currentGroupIndex
+                    && allDay === currentAllDay
+                    && this._compareDatesAndAllDay(date, currentStartDate, currentEndDate, allDay)) {
+                    return {
+                        position: {
+                            columnIndex,
+                            rowIndex: showAllDayPanel && !isVerticalGroupOrientation
+                                ? rowIndex - 1
+                                : rowIndex,
+                        },
+                        cellData,
+                    };
+                }
+            }
+        }
+
+        return undefined;
```

Comment:
Do we realy need to return 'undefined' ?

Reply:
Yes. It may return undefined if the cell is not in the viewport - in case we select a lot fo cells in several days. And, as it seems to me, it is better to reflect this in code as `return undefined`